### DISCUSSION TITLE
Proposal: add a feature to publish NuGet packages

### DISCUSSION THREAD

**Discussion by samsmithnz** [isaacrlevin/WorkFlowGenerator#6] [Ideas]
- would include versioning with GitVersion (can we add dependent files?) [example](https://github.com/samsmithnz/AzurePipelinesToGitHubActionsConverter/blob/main/.github/workflows/CI.yml)
- packing to create the nupkg
- publishing to NuGet and/or GitHub packages
- create a release to lock in the version/tag to the repo

**Comment by isaacrlevin** [comment]
I love this idea! How would you discern if the package needed to be published to Nuget/GH Packages? I know Nuget has some special stuff in the csproj, but does GitHub?

**Reply by samsmithnz** [reply]
There is some special stuff in the project file, but it's the same for both GH and NuGet. The tool could scan the project file to see if said special stuff exists (or another flag to ignore it). Let me spec it out here as a bit of a proposal (parameter names are not final - and are extra verbose to demonstrate purpose for these docs)

# NuGet workflow enhancement

**Goal**: Create a workflow based on a target .NET library, with options to target source, and/or detect if project is setup correctly. 


```
dotnet-workflow-generator [<PROJECT>] 
    [-c|--configuration <CONFIGURATION>]
    [--nuget-source <NuGet source (NuGet.org/GitHub packages)>] 
    [--include-git-version] 
    [--include-symbols]
    [--check-project-for-symbol-packaging-format]
```

for example, to publish a .NET project to NuGet.org, 
```
dotnet-workflow-generator C:\dev\somefolderwithaprojectfile 
    --configuration Release
    --nuget-source "https://api.nuget.org/v3/index.json" 
    --include-git-version 
    --check-project-for-symbol-packaging-format
```

This would create a workflow with information to version, pack, and publish a NuGet package to NuGet.org.
